egs/librispeech/ASR/conformer_ctc_madam_no_warmup/train.py

```diff
@@ -0,0 +1,767 @@
+#!/usr/bin/env python3
+
+# This is just at the very beginning ...
+
+import argparse
+import logging
+from pathlib import Path
+from shutil import copyfile
+from typing import Optional
+
+import k2
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import torch.nn as nn
+from conformer import Conformer
+from lhotse.utils import fix_random_seed
+
+#  from transformer import Noam
+from madam import Moam
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.nn.utils import clip_grad_norm_
+from torch.utils.tensorboard import SummaryWriter
+
+from icefall.bpe_graph_compiler import BpeCtcTrainingGraphCompiler
+from icefall.checkpoint import load_checkpoint
+from icefall.checkpoint import save_checkpoint as save_checkpoint_impl
+from icefall.dataset.librispeech import LibriSpeechAsrDataModule
```

Comment:
I strongly suggest making this class local too (e.g. in a local `data.py` file) -- in the current repo layout, it will make it much easier to experiment with different types of data setups and augmentations.